cmd/root.go

```diff
@@ -23,22 +23,23 @@ import (
 	"sort"
 	"strings"
 
-	"github.com/spf13/cobra"
-	"sigs.k8s.io/release-utils/version"
-
 	"github.com/ossf/scorecard/v4/checker"
 	"github.com/ossf/scorecard/v4/clients"
-	docs "github.com/ossf/scorecard/v4/docs/checks"
-	sclog "github.com/ossf/scorecard/v4/log"
+	"github.com/ossf/scorecard/v4/dependencydiff"
 	"github.com/ossf/scorecard/v4/options"
 	"github.com/ossf/scorecard/v4/pkg"
 	"github.com/ossf/scorecard/v4/policy"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/release-utils/version"
+
+	docs "github.com/ossf/scorecard/v4/docs/checks"
+	sclog "github.com/ossf/scorecard/v4/log"
 )
 
 const (
 	scorecardLong = "A program that shows security scorecard for an open source software."
 	scorecardUse  = `./scorecard (--repo=<repo> | --local=<folder> | --{npm,pypi,rubygems}=<package_name>)
-	 [--checks=check1,...] [--show-details]`
+	 [--checks=check1,...] [--show-details] [--dependencydiff=<base>...<head>]`
```

Comment:
Any reason we decided to do this as a flag instead of sub-command?

Reply:
that was not discussed, and I wanted some feedback on that. Thanks

Reply:
@azeemshaikh38 @laurentsimon @naveensrinivasan 
sync from chat:
justifications of using a sub-command:

(1) it's easier to maintain redundant flag names than validation code for all different combinations of flags

(2) although we don't have sub-commands for Scorecard now, cobra has support for it and we can easily add a scorecard dependencydiff to separate the original Scorecard running mode (running checks on the current repo) and the dependencydiff running mode. (https://cobra.dev/)

(3) it doesn't affect the reuse of other flags, such as re-using --repo to get the repo. those still work for sub-commands